connect_ext_test.go

```diff
@@ -714,6 +716,202 @@ func TestInterceptorReturnsWrongType(t *testing.T) {
 	assert.True(t, strings.Contains(connectErr.Message(), "unexpected client response type"))
 }
 
+func TestHandlerWithReadMaxBytes(t *testing.T) {
+	t.Parallel()
+	mux := http.NewServeMux()
+	readMaxBytes := 1024
+	mux.Handle(pingv1connect.NewPingServiceHandler(
+		pingServer{},
+		connect.WithReadMaxBytes(readMaxBytes),
+	))
+	readMaxBytesMatrix := func(t *testing.T, client pingv1connect.PingServiceClient, compressed bool) {
+		t.Helper()
+		t.Run("equal_read_max", func(t *testing.T) {
+			t.Parallel()
+			// Serializes to exactly readMaxBytes (1024) - no errors expected
```

Comment:
Excellent, glad you're testing this - looking at your code, I bet my previous implementation had an off-by-one error.